lesson18/index.js

```diff
@@ -0,0 +1,197 @@
+}
+
+function init(data) {
+    const allofCountElement = document.getElementById("js-counter-all");
+    
+    allofCountElement.textContent = data.length;
+    incrementCurrentIndex();
+    toggleButtonDisabled(data.length);
+}
+
+function toggleButtonDisabled(length) {
+    const currentIndex = getCurrentIndex();
+    const firstIndex = 0;
+    const lastIndex = length - 1;
+
+    previousButton.disabled = currentIndex === firstIndex;
+    nextButton.disabled = currentIndex === lastIndex;
+}
+
+function getCurrentIndex() {
+    const activeImg = ul.querySelector(".is-active");
+    return Number(activeImg.dataset.index);
+}
+
+function incrementCurrentIndex() {
+    const currentCountElement = document.getElementById("js-counter-current");
+    currentCountElement.textContent = getCurrentIndex() + 1;
+}
+
+function switchImg(direct) {
+    const activeImg = ul.querySelector(".is-active");
+    activeImg.classList.remove("is-active");
+    activeImg[direct].classList.add("is-active");
+}
+
+function switchDotPagination(direct) {
+    const ul = document.getElementById("js-pagination-list");
+    const activeDot = ul.querySelector(".is-active");
+    activeDot.classList.remove("is-active");
+    activeDot[direct].classList.add("is-active");
+}
+
+const addEventListenerForNextButton = (length) => {
+    nextButton.addEventListener ("click", () => {
+        switchImg("nextElementSibling");
+        switchDotPagination("nextElementSibling");
+        incrementCurrentIndex();
+        toggleButtonDisabled(length);
+    })
+}
+
+const addEventListenerForPreviousButton = (length) => {
+    previousButton.addEventListener ("click", () => {
+        switchImg("previousElementSibling");
+        switchDotPagination("previousElementSibling");
+        incrementCurrentIndex();
+        toggleButtonDisabled(length);
+    })
+}
+
+const addEventListenerForDotPagination = (length) => {
+    const dotPaginationList = document.getElementById("js-pagination-list");
+
+    dotPaginationList.addEventListener ("click", (e) => {
+
+        //buttonとbuttonの間はクリック対象外にする
+        if (dotPaginationList && e.currentTarget !== e.target) {
+            const clickedDotIndex = e.target.dataset.index;
+
+            //dotを切り替える
+            const activeDot = dotPaginationList.querySelector(".is-active");
+            const paginationItems = Array.from(document.getElementsByClassName("js-pagination-item"));
+            activeDot.classList.remove("is-active");
+            paginationItems[clickedDotIndex].classList.add("is-active");
+
+            //数字のページネーションを切り替える
+            const currentCountElement = document.getElementById("js-counter-current");
+            currentCountElement.textContent = Number(clickedDotIndex) + 1;
+
+            //画像を切り替える
+            const activeImg = ul.querySelector(".is-active");
+            const imgItems = Array.from(document.getElementsByClassName("js-slideshow-item"));
```

Comment:
IMO
You can do the same thing with Spread syntax, so try it!
[https://developer.mozilla.org/ja/docs/Web/JavaScript/Reference/Operators/Spread_syntax](スプレッド構文)
```suggestion
            const imgItems =[...document.getElementsByClassName("js-slideshow-item")];
```

Reply:
Comment
Personally, each process of this function is straightforward, 
so I didn't think it was necessary to cut it out.

However, in other functions, the processing is cut out in detail for each function, 
so it may be better to cut it in consideration of the consistency of the entire code.

Reply:
@443n0511 
Thanks for letting me know!
In this case, the spread syntax is more intuitive and easy to understand that it is an array!
Fixed!
[d3286bb](https://github.com/haru-programming/JavaScript-lessons/pull/22/commits/d3286bbcb73be1c507af1f1610d3c47c4821ab78)

Reply:
@nakagawamai 
Thank you for your comments! 

> so it may be better to cut it in consideration of the consistency of the entire code.

I feel that way myself!
So I will try to cut it out as a function. I appreciate your reading the code and giving me your honest opinion!
Thank you :) !!